File: app/src/main/java/com/guuguo/learnsave/net/retrofit/GankService.java
package com.guuguo.learnsave.net.retrofit;

import com.guuguo.learnsave.bean.GankDays;
import com.guuguo.learnsave.bean.Ganks;

import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Created by gaohailong on 2016/5/17.
 */
public interface GankService {
    @GET("data/{type}/{count}/{page}")
    Observable<Ganks> getGanHuo(@Path("type") String type, @Path("count") int count, @Path("page") int page);

    @GET("day/{date}")
    Observable<GankDays> getGankOneDay(@Path("date") String date);
    
    @GET("day/{history}")
    Observable<GankDays> getGankDates();
}

File: app/src/main/java/com/guuguo/gank/net/GankService.kt
package com.guuguo.gank.net

import com.guuguo.gank.model.GankDays
import com.guuguo.gank.model.Ganks
import com.guuguo.gank.model.GankNetResult
import com.guuguo.gank.model.entity.GankModel
import io.reactivex.Observable
import java.util.*

/**
 * Created by gaohailong on 2016/5/17.
 */
interface GankService {
    @retrofit2.http.GET("data/{type}/{count}/{page}")
    fun getGanHuo(@retrofit2.http.Path("type") type: String, @retrofit2.http.Path("count") count: Int, @retrofit2.http.Path("page") page: Int)
            : Observable<Ganks<ArrayList<GankModel>>>

    @retrofit2.http.GET("day/{date}")
    fun getGankOneDay(@retrofit2.http.Path("date") date: String): io.reactivex.Single<GankDays>

    @retrofit2.http.GET("search/query/{query}/category/{category}/count/{count}/page/{page} ")
    fun getGankSearchResult(@retrofit2.http.Path("query") query: String, @retrofit2.http.Path("category") category: String, @retrofit2.http.Path("count") count: Int, @retrofit2.http.Path("page") page: Int): io.reactivex.Single<GankNetResult>
}
